tests/integration/test_configuration.py

```diff
@@ -0,0 +1,231 @@
+                "project_id": "{{ ZZ_INTEGRATION_PROJECT_ID }}",
+                "module_hierarchy": [
+                  "{{ RPC_PRODUCT_RELEASE }}",
+                  "{{ RPC_RELEASE }}",
+                  "{{ JOB_NAME }}",
+                  "{{ MOLECULE_TEST_REPO }}",
+                  "{{ zz_testcase_class }}"
+                ]
+              }
+            }"""
+
+    with open(filename, 'w') as f:
+        f.write(config_json)
+
+    return filename
+
+
+@pytest.fixture(scope='session')
+def no_module_hierarchy_config(tmpdir_factory):
+    """A configuration missing the test_cycle config
+    """
+
+    filename = tmpdir_factory.mktemp('data').join('config_file.json').strpath
+    config_json = \
+        """{
+              "zigzag": {
+                "test_cycle": "{{ ZZ_INTEGRATION_TEST_CYCLE }}",
+                "project_id": "{{ ZZ_INTEGRATION_PROJECT_ID }}"
+              }
+            }"""
+
+    with open(filename, 'w') as f:
+        f.write(config_json)
+
+    return filename
+
+
+@pytest.fixture(scope='session')
+def no_project_id_config(tmpdir_factory):
+    """A configuration missing the test_cycle config
+    """
+
+    filename = tmpdir_factory.mktemp('data').join('config_file.json').strpath
+    config_json = \
+        """{
+              "zigzag": {
+                "test_cycle": "{{ ZZ_INTEGRATION_TEST_CYCLE }}",
+                "module_hierarchy": [
+                  "{{ RPC_PRODUCT_RELEASE }}",
+                  "{{ RPC_RELEASE }}",
+                  "{{ JOB_NAME }}",
+                  "{{ MOLECULE_TEST_REPO }}",
+                  "{{ zz_testcase_class }}"
+                ]
+              }
+            }"""
+
+    with open(filename, 'w') as f:
+        f.write(config_json)
+
+    return filename
+
+
+@pytest.fixture(scope='session')
+def required_configs(tmpdir_factory):
+    """A configuration with required configs
+    used to test when properties evaluate to empty things
+    """
+
+    filename = tmpdir_factory.mktemp('data').join('config_file.json').strpath
+    config_json = \
+        """{
+              "zigzag": {
+                "test_cycle": "{{ ZZ_INTEGRATION_TEST_CYCLE }}",
+                "project_id": "{{ ZZ_INTEGRATION_PROJECT_ID }}",
+                "module_hierarchy": [
+                  "{{ ZZ_MODULE_HIERARCHY_ONE }}",
+                  "{{ ZZ_MODULE_HIERARCHY_TWO }}"
+                ]
+              }
+            }"""
+
+    with open(filename, 'w') as f:
+        f.write(config_json)
+
+    return filename
+
+
+# ======================================================================================================================
+# Test Suites
+# ======================================================================================================================
+class TestConfig(object):
+    # noinspection PyUnresolvedReferences
+    def test_publish_single_passing_test_with_custom_mod_hierarchy(self, single_passing_test_with_custom_mod_hierarchy):
```

Comment:
It looks like these tests are both using values that are hard coded in.  I think a better test would be if we altered this test to have the module hierarchy change based on what properties are present.  That seems like it would be a good equivalent for [the rename tests that we are skipping](https://github.com/rcbops/zigzag/blob/configuration_file/tests/integration/test_basic.py#L249-L250) 

Here is what I am thinking for the config value for `module_hierarchy` 
` "module_hierarchy": ["{{ FOO if FOO else BAR }}", "{{ zz_testcase_class }}"]`

This way you can can change the module_hierarchy by adding or removing props.  If this was done in one test we could eliminate all the skipped tests.

Reply:
the test added in b4b315c6adad9db9e080d1edc3aba6a90d351b86 is currently failing when I don't think it should.  I'd like to discuss this further @zreichert

Reply:
I have filed https://rpc-openstack.atlassian.net/browse/ASC-1654 , I'll have a PR up for this today.

Reply:
I think the spirit of the [skipped tests ](https://github.com/rcbops/zigzag/blob/configuration_file/tests/integration/test_basic.py#L249-L417) had two parts.  

1. run zigzag & validate its module hierarchy
2. run zigzag again with the same test and validate that its module hierarchy had changed

@ryan-rs You originally wrote these tests.  Is that the spirit of what is happening in these tests?

[These fixtures](https://github.com/rcbops/zigzag/blob/a8adc6bced9c1397ded2633ff5baeedd10453401/tests/integration/test_configuration.py#L161-L210) look to be the same (one flasey one truthy).  If you change the `False` & `True` to a named property XYZ.

Change
`zz_runner = _zigzag_runner_factory('junit.xml', computed_hierarchy_config_file_truthy, {})`

to 

`zz_runner = _zigzag_runner_factory('junit.xml', computed_hierarchy_config_file_truthy, {"XZY": "Well hello there"})`

Reply:
@zreichert Yes, the idea was to verify that the results respect the hierarchy.